fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/QuerySegmentAggregator.java

```diff
@@ -633,19 +640,25 @@ protected void buildWhereClause(StringBuilder whereClause, String overrideType)
                                             .append(" AND LR.LOGICAL_RESOURCE_ID = ")
                                             .append(paramTableAlias)
                                             .append(".LOGICAL_RESOURCE_ID");
-                            }
-                        }
+                            } // end if/else for NOT / NOT-IN
+                        } // end if/else for isReverseChained
                     } else {
                         whereClause.append(querySegment.getQueryString()
                             .replaceAll(PARAMETER_TABLE_ALIAS + "_p", paramTableAlias + "_p"));
-                    }
-                }
+                        whereWordAdded = true;
+                    } // end if/else for COMPOSITE
+                } // end if/else for MISSING
             } // end if SKIP_WHERE
         } // end for
 
         // If there were any query parameters with :missing, :not, or :not-in modifier, append the missingOrNotModifierWhereClause
         if (missingOrNotModifierWhereClause.length() > 0) {
             whereClause.append(missingOrNotModifierWhereClause.toString());
+            whereWordAdded = true;
+        }
+
+        if (!SortedQuerySegmentAggregator.class.equals(this.getClass())) {
```

Comment:
this was one of the hackiest lines.  we probably don't want to merge this...
or if we do merge it, it would just be a stopgap til we have a more elegant solution

Reply:
it was needed because the SortedQuerySegmentAggregator actually adds a JOIN clause *after* calling this buildWhereClause method...so in the case of SortedQuerySegmentAggregator I add this NOT_DELETED clause after that instead.